src/plugin/timezone/index.js

```diff
@@ -18,21 +27,29 @@ const getDateTimeFormat = (timezone, options = {}) => {
   let dtf = dtfCache[key]
   if (!dtf) {
     dtf = new Intl.DateTimeFormat('en-US', {
+      ...dateTimeFormatDefaults,
       hour12: false,
       timeZone: timezone,
-      year: 'numeric',
-      month: '2-digit',
-      day: '2-digit',
-      hour: '2-digit',
-      minute: '2-digit',
-      second: '2-digit',
       timeZoneName
     })
     dtfCache[key] = dtf
   }
   return dtf
 }
 
+const localeStringifierCache = {}
+const getLocaleStringifier = (timezone) => {
+  let localeStringifier = localeStringifierCache[timezone]
+  if(!localeStringifier) {
+    localeStringifier = new Intl.DateTimeFormat('en-US', {
+      ...dateTimeFormatDefaults,
+      timeZone: timezone,
+    })
+    localeStringifierCache[timezone] = localeStringifier
+  }
+  return localeStringifier
+}
+
```

Comment:
should the cache directly be part of the function, its the usual approach for memoization in js

```ts
const getLocaleStringifier = (() => {
  const localeStringifierCache = {};
  return (timezone) => {
    // Check if the locale stringifier is cached
    if (!localeStringifierCache[timezone]) {
      // Create new locale stringifier if not cached
      localeStringifierCache[timezone] = new Intl.DateTimeFormat('en-US', {
        ...dateTimeFormatDefaults,
        timeZone: timezone,
      });
    }

    // Return locale stringifier from cache
    return localeStringifierCache[timezone];
  };
})()
```

Reply:
might not change anything in this context not sure